core/util.js

```diff
@@ -75,10 +75,67 @@ function getStatisticName (alarmConfig) {
   return alarmConfig.Statistic || alarmConfig.ExtendedStatistic
 }
 
+/**
+ * Given a CloudFormation resource for an API Gateway REST API, derive CloudFormation syntax for
+ * the API name.
+ * The API name can be provided as a `Name` property or in the OpenAPI specification as
+ * `Body.info.title`
+ *
+ * In either case, the name can be a string literal or use a CloudFormation intrinsic function
+ * (Sub, Ref or GetAtt)
+ *
+ * @param restApiResource CloudFormation resource for a REST API
+ * @param restApiLogicalId The logical ID for the resouce
+ * @returns CloudFormation syntax for the API name
+ */
+function resolveRestApiNameAsCfn (restApiResource, restApiLogicalId) {
+  if (restApiResource.Properties.Name) {
+    return restApiResource.Properties.Name
+  }
+  if (restApiResource.Properties.Body && restApiResource.Properties.Body.info && restApiResource.Properties.Body.info.title) {
+    return restApiResource.Properties.Body.info.title
+  }
+  throw new Error(`No API name specified for REST API ${restApiLogicalId}. Either Name or Body.info.title should be specified`)
```

Comment:
This should be tested but I think we could simplify it a little as follows (if you think this is more readable):

```suggestion
  const apiName = restApiResource.Properties.Name || restApiResource.Properties?.Body?.info?.title
  
  if (!apiName) {
    throw new Error(`No API name specified for REST API ${restApiLogicalId}. Either Name or Body.info.title should be specified`)
  }
  
  return apiName  
```

Reply:
We still have Node.js 12 support in this branch. If we removed Node 12 support, that's fine, but would probably warrant consistent use of `?.` throughout the code then.

Reply:
changed